### PR TITLE
Code for viewing common leads in two or more lists.

### PR DESCRIPTION
**Pull request by ps92**
Feature code for issue https://github.com/mautic/mautic/issues/1440.

### CODE REVIEW

**Comment by ps92** [comment]
I need a little help. I wrote the query in native SQL. I would be really thankful if someone could help me write the same for QueryBuilder. 

Thank you.

**Comment by ps92** [comment]
This feature would basically provide the common people in two or more lists. 
If we needed for two or more lists to be exclusive with no conflicts, this feature would help us.

File: app/bundles/LeadBundle/Entity/LeadRepository.php [hunk 1]
```diff
@@ -799,6 +800,54 @@ protected function addSearchCommandWhereClause(&$q, $filter)
...
+
+
+                $imploder = implode(",", $imploder);
+                
+                $return = array();
+                $rsm = new ResultSetMapping();
+
+
+
+                $query_ps = "SELECT x.lead_id FROM (SELECT *, COUNT(lead_id) FROM lead_lists_leads WHERE leadlist_id IN (".$imploder.") GROUP BY lead_id  HAVING COUNT(lead_id) > ".$pluck." AND SUM(manually_removed) < ".$pluck.") AS X";
```
Comment (kuzmany) [inline app/bundles/LeadBundle/Entity/LeadRepository.php]: MAUTIC_TABLE_PREFIX missing

Comment (ps92) [inline app/bundles/LeadBundle/Entity/LeadRepository.php]: Thanks @kuzmany . It helped.

**Comment by ps92** [comment]
If you typed "common:{alias1}+{alias2}" in the search bar in Manage Leads page, you can see the leads common in the two lists. 
You can specify N number of aliases, each separated by a "+" sign.

File: app/bundles/LeadBundle/Entity/LeadRepository.php [hunk 2]
```diff
@@ -799,6 +800,54 @@ protected function addSearchCommandWhereClause(&$q, $filter)
...
+
+
+                $imploder = implode(",", $imploder);
+                
+                $return = array();
+                $rsm = new ResultSetMapping();
+
+
+
+                $query_ps = "SELECT X.lead_id FROM (SELECT *, COUNT(lead_id) FROM ".MAUTIC_TABLE_PREFIX."lead_lists_leads WHERE leadlist_id IN (".$imploder.") GROUP BY lead_id  HAVING COUNT(lead_id) > ".$pluck." AND SUM(manually_removed) < ".$pluck.") AS X";
```
Comment (dongilbert) [inline app/bundles/LeadBundle/Entity/LeadRepository.php]: I like the PR here. Can you update this line to use the Doctrine query builder?

**Comment by ps92** [comment]
@dongilbert Sorry! I am a bit late, but I had almost forgotten about this. 
I converted this to Doctrine's query builder.

**Comment by ps92** [comment]
@dbhurley Can you please review this code? There are conflicts too now on this commit.

**Comment by ps92** [comment]
I have pushed new changes. This should resolve the conflicts. 
`https://github.com/mautic/mautic/pull/2599/commits`

**Comment by ps92** [comment]
#2599

**Comment by alanhartless** [comment]
Thanks - closing in favor of #2599.

**Pull request closed without merging**